README.md

```diff
@@ -66,3 +66,111 @@ that can be used to quickly bring up a GKE cluster and configure
+The output from this `helm template` command is then piped to [yq](https://pypi.org/project/yq/). 
+`yq` allows us to pull out just the information we're interested in, either by referencing its position in the yaml file directly or giving information about it (like its length).
+
+The test passes or fails based on the conditional at the end that is in square brackets, which is a comparison of our expected value and the output of  `helm template` piped to `yq`.
+
+The `| tee /dev/stderr ` pieces direct any terminal output of the `helm template` and `yq` commands to stderr so that it doesn't interfere with `bats`.
+
+#### Test Examples
+
+Here are some examples of common test patterns:
+
+- Check that a value is disabled by default
+
+    ```
+    @test "ui/Service: no type by default" {
+      cd `chart_dir`
+      local actual=$(helm template \
+          -x templates/ui-service.yaml  \
+          . | tee /dev/stderr |
+          yq -r '.spec.type' | tee /dev/stderr)
+      [ "${actual}" = "null" ]
+    }
+    ```
+
+    In this example, nothing is changed from the default templates (no `--set` flags), then we use `yq` to retrieve the value we're checking, `.spec.type`.
+    This output is then compared against our expected value (`null` in this case) in the assertion `[ "${actual}" = "null" ]`
+
+
+- Check that a template value is rendered to a specific value
+    ```
+    @test "ui/Service: specified type" {
+      cd `chart_dir`
+      local actual=$(helm template \
+          -x templates/ui-service.yaml  \
+          --set 'ui.service.type=LoadBalancer' \
+          . | tee /dev/stderr |
+          yq -r '.spec.type' | tee /dev/stderr)
+      [ "${actual}" = "LoadBalancer" ]
+    }
+    ```
+
+    This is very similar to the last example, except we've changed a default value with the `--set` flag and correspondingly changed the expected value.
+
+ - Check that a template value contains several values
+    ```
+    @test "syncCatalog/Deployment: to-k8s only" {
+      cd `chart_dir`
+      local actual=$(helm template \
+          -x templates/sync-catalog-deployment.yaml  \
+          --set 'syncCatalog.enabled=true' \
+          --set 'syncCatalog.toConsul=false' \
+          . | tee /dev/stderr |
+          yq '.spec.template.spec.containers[0].command | any(contains("-to-consul=false"))' | tee /dev/stderr)
+      [ "${actual}" = "true" ]
+
+      local actual=$(helm template \
+          -x templates/sync-catalog-deployment.yaml  \
+          --set 'syncCatalog.enabled=true' \
+          --set 'syncCatalog.toConsul=false' \
+          . | tee /dev/stderr |
+          yq '.spec.template.spec.containers[0].command | any(contains("-to-k8s"))' | tee /dev/stderr)
+      [ "${actual}" = "false" ]
+    }
+    ```
+    In this case, the same command is run twice in the same test.
+    This can be used to look for several things in the same field, or to check that something is not present that shouldn't be.
```

Comment:
Is there a reason to do it like that vs something like:

```sh
local actualYAML=$(helm template \
          -x templates/sync-catalog-deployment.yaml  \
          --set 'syncCatalog.enabled=true' \
          --set 'syncCatalog.toConsul=false' \
          . | tee /dev/stderr)

local notToConsul=$(echo "actualYAML" | \
    yq '.spec.template.spec.containers[0].command | any(contains("-to-consul=false"))' | tee /dev/stderr)
[ "${notToConsul}" = "true" ]

local actual=$(helm template \
          -x templates/sync-catalog-deployment.yaml  \
          --set 'syncCatalog.enabled=true' \
          --set 'syncCatalog.toConsul=false' \
          . | tee /dev/stderr)
local containsToK8s=$(yq '.spec.template.spec.containers[0].command | any(contains("-to-k8s"))' | tee /dev/stderr)
[ "${containsToK8s}" = "false" ]
```

No big deal just seems like it would be marginally more efficient and less copy paste to just run once and make multiple assertions on the output?

Reply:
(I imagine the second `actual` definition was a copy/paste slip, since as written this adds additional work rather than removing it.)

In this case, I don't think the minimal redundancy reduction makes it worth changing the established pattern. For many contributors, this will be their first time working with `bats`, so keeping things as consistent as possible will reduce the overhead to understanding what's going on. In cases where more than just two conditions being tested on the same `helm template` output, combining the base command would definitely be preferable. I'll make a note of that in the comment for this example.